### PR TITLE
Add support for running pyside2-uic binary to dynamically compile ui files

### LINKED ISSUE #1102: pyqtgraph.Qt.loadUiType broken for PySide2 >= 5.14
https://dev.azure.com/pyqtgraph/pyqtgraph/_build/results?buildId=557&view=logs&j=1d023af0-a188-58ef-16b4-2e987dd32dd1&t=addebc46-66b7-583c-f559-d6f453020f3e&l=308

There's a note here, though I'm not sure what it means precisely: https://wiki.qt.io/Qt_for_Python_Development_Notes#24._October_2019

Location of the issue for reference: https://github.com/pyqtgraph/pyqtgraph/blob/6e9a48558c010803883f0f83a7170581098b978a/pyqtgraph/Qt.py#L91

### PR DESCRIPTION
**Pull request by campagnola**
fixes #1102

### CODE REVIEW

**Comment by j9ac9k** [comment]
Hmm, all linux CI runs are failing, but not just on this PR, but on another one as well... wondner if there is some upstream issue.

**Comment by j9ac9k** [comment]
Closing/Opening to re-trigger CI run

**Comment by 2xB** [comment]
#1102 is closed, does that affect this PR?

**Comment by j9ac9k** [comment]
@2xB that was my fault, I didn't verify the solution worked entirely and closed the issue, turns out we do need a change in the pyqtgraph  code-base

File: pyqtgraph/Qt.py [hunk 1]
```diff
@@ -119,6 +119,8 @@ def _loadUiType(uiFile):
 
     # convert ui file to python code
     if pysideuic is None:
+        if PySide2.__version__[:5].split('.')[:2] == ['5', '14']:
+            warnings.warn('For UI compilation, it is recommended to upgrade to PySide >= 5.15')
```
Comment (j9ac9k) [inline pyqtgraph/Qt.py:123]: Could make this be `>= 5.14.2.2`

Comment (j9ac9k) [reply pyqtgraph/Qt.py:123]: or better yet.

```python
pysideVersion = tuple(map(int, PySide2.__version.split(".")))
if pysideVersion >= (5, 14) and pysideVersion < (5, 14, 2, 2):
    warnings.warn(....)
```

**Comment by j9ac9k** [comment]
I think this branch is ready to merge @campagnola ... figured we should at least test the new loadUiType functionality in the CI platform, so I added a pyside2-latest pipeline (now up to 18, yay!)

**Comment by campagnola** [comment]
Thanks @j9ac9k !

**Pull request merged**